cudaungappedextender/include/claraparabricks/genomeworks/cudaungappedextender/cudaungappedextender.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#pragma once
+
+namespace claraparabricks
+{
+namespace genomeworks
+{
+namespace cudaungappedextender
+{
+
+struct Seed
```

Comment:
`Seed` is the wrong name in the context of WGA. We can either name it `SeedHit` or revert back to the Genomeworks usage of `Anchor`.

Reply:
`Anchor` in Genomeworks currently has references to the query and target sequences in that structure. `SeedHit` implies that there's a missing step in this offering which should generate the `Hits` from `Seeds`, which is indeed a step in segalign. However, do the seedHits there not form seed values for the isolated scope of the ungapped extender?

Reply:
No, `Seeds` are just the subsequences in the query sequence. `SeedHit` is the target+query subsequence. Maybe we have `segmentPair` in place of `Seed` and `HSP` in place of `segmentPair`.

Reply:
There's no length associated with the `SeedHit/SegmentPair/Seed` data structure. Is it valid to call it a subsequence/segment in this generalized context (putting aside the segalin usecase)? Also I'm trying to avoid adding the `HighScore/HighScoring` label to current `ScoredSegment` (which can be named as `ScoredSegmentPair` or `SegmentPair`) as it should be apparent from the `score_threshold` parameter that is passed in to keep it general.

Reply:
Let's just make it `SeedPair` (seed length is the implied length) and `ScoredSegmentPair` then?

Reply:
Yes. That works!